pkg/kubelet/metrics/collectors/resource_metrics.go

```diff
@@ -31,56 +31,56 @@ var (
 		"Cumulative cpu time consumed by the node in core-seconds",
 		nil,
 		nil,
-		metrics.ALPHA,
+		metrics.STABLE,
 		"")
 
 	nodeMemoryUsageDesc = metrics.NewDesc("node_memory_working_set_bytes",
 		"Current working set of the node in bytes",
 		nil,
 		nil,
-		metrics.ALPHA,
+		metrics.STABLE,
 		"")
 
 	containerCPUUsageDesc = metrics.NewDesc("container_cpu_usage_seconds_total",
 		"Cumulative cpu time consumed by the container in core-seconds",
 		[]string{"container", "pod", "namespace"},
 		nil,
-		metrics.ALPHA,
+		metrics.STABLE,
 		"")
 
 	containerMemoryUsageDesc = metrics.NewDesc("container_memory_working_set_bytes",
 		"Current working set of the container in bytes",
 		[]string{"container", "pod", "namespace"},
 		nil,
-		metrics.ALPHA,
+		metrics.STABLE,
 		"")
 
 	podCPUUsageDesc = metrics.NewDesc("pod_cpu_usage_seconds_total",
 		"Cumulative cpu time consumed by the pod in core-seconds",
 		[]string{"pod", "namespace"},
 		nil,
-		metrics.ALPHA,
+		metrics.STABLE,
 		"")
 
 	podMemoryUsageDesc = metrics.NewDesc("pod_memory_working_set_bytes",
 		"Current working set of the pod in bytes",
 		[]string{"pod", "namespace"},
 		nil,
-		metrics.ALPHA,
+		metrics.STABLE,
 		"")
 
 	resourceScrapeResultDesc = metrics.NewDesc("scrape_error",
 		"1 if there was an error while getting container metrics, 0 otherwise",
 		nil,
 		nil,
-		metrics.ALPHA,
+		metrics.STABLE,
 		"")
```

Comment:
This metric is very odd, I don't think we should graduate it to stable as is. Having the error as a 0/1 gauge will hide errors happening outside of the scrape interval or when scraping the /metrics/resource endpoint fails.

@logicalhan what do you think?

Reply:
What's the frequency which we fetch container metrics? IIRC, it was something like 30 seconds, which means this gauge is actually fine since scrape intervals tend to be drastically smaller than 30 seconds.

Reply:
I don't remember, I would have to check, but it should be smaller than that since metrics-server scrape interval on this endpoint is 15s.

But even if the scrape interval is similar, what if multiple scrape fails for a certain period of time? We would lose that info, whereas if we had a counter we wouldn't.

Reply:
@dgrisonnet @logicalhan could you please remind me - how to deprecate `scrape_error` in favor of `resource_scrape_error` ? I have introduced a STABLE `resource_scrape_error` metric, but what to do with `scrape_error`? Keep it as ALPHA and deprecate in future? Or still promote it to STABLE and then deprecate it ?